nova/virt/vmwareapi/session.py

```diff
@@ -58,10 +100,34 @@ def _call_method(self, module, method, *args, **kwargs):
         """Calls a method within the module specified with
         args provided.
         """
-        if not self._is_vim_object(module):
-            return self.invoke_api(module, method, self.vim, *args, **kwargs)
+        try:
+            if not self._is_vim_object(module):
+                return self.invoke_api(module, method, self.vim, *args,
+                                       **kwargs)
 
-        return self.invoke_api(module, method, *args, **kwargs)
+            return self.invoke_api(module, method, *args, **kwargs)
+        except vexc.ManagedObjectNotFoundException as monfe:
+            with excutils.save_and_reraise_exception() as ctxt:
+                moref = monfe.details.get("obj") if monfe.details else None
+                for arg in itertools.chain(args, six.itervalues(kwargs)):
+                    if not isinstance(arg, StableMoRefProxy):
+                        continue
+                    moref_arg = get_moref_value(arg.moref)
```

Comment:
I thought, because we know the type of `arg`, and know that it has the attribute `moref,` we can safe the step going through the proxy, and it will also fail, if it isn't.